### DISCUSSION TITLE
Why does the "get_phonons_in_qpath" plot strange phonon dispersion?

### DISCUSSION THREAD

**Discussion by biglinn** [SSCHAcode/python-sscha#352] [Q&A]
Hi, @mesonepigreco:
I find a question about plotting the phonon dispersion.
When I plot the phonon dispersion with `plot_dispersion.py`(main codes are as follows), it shows strange imaginary frequency:
```
dyn = CC.Phonons.Phonons("dyn_mat/sscha_dyn_population10_", 6)

qpath, data = CC.Methods.get_bandpath(dyn.structure.unit_cell,
                                      PATH,
                                      SPECIAL_POINTS,
                                      N_POINTS)
xaxis, xticks, xlabels = data # Info to plot correclty the x axis

dispersion = CC.ForceTensor.get_phonons_in_qpath(dyn, qpath)
```
![population40](https://github.com/user-attachments/assets/259b85f5-9c6f-4484-9d62-f6460140158a)

However, I get `dyn0`(the file containing all irredicuble q-points, q2r.x needs it) back, then use q2r.x and matdyn.x to plot the dispersion(asr is simple).
In this case, the dispersion is as follows:
![population40_q2r](https://github.com/user-attachments/assets/0bd7e0a9-98d1-404f-9754-70bd51507fde)

I wonder if the `get_phonons_in_qpath` has any limitations.

**Comment by mesonepigreco** [comment]
This seems suspicious.
In particular, you have an imaginary phonon mode at Gamma, but this point should not be interpolated. The ZU part of your dispersion is substantially different. 

Can you upload the dynamical matrix so that I can try to reproduce the error? 
Thanks a lot,
LM

**Reply by biglinn** [reply]
Thank you for your help!
[dyn.zip](https://github.com/user-attachments/files/16430510/dyn.zip)
These are the dynamical matrix(dyn1~8). The "dyn0" is get from ph.x, containing all irredicuble q-points.
